src/PrestaShopBundle/Controller/Admin/Improve/International/StateController.php

```diff
@@ -64,4 +81,302 @@ public function getStatesAction(Request $request)
+     *     redirectRoute="admin_states_index",
+     *     message="You do not have permission to delete this."
+     * )
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function deleteBulkAction(Request $request): RedirectResponse
+    {
+        $stateIds = $this->getBulkStatesFromRequest($request);
+
+        try {
+            $this->getCommandBus()->handle(new BulkDeleteStatesCommand($stateIds));
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion.', 'Admin.Notifications.Success')
+            );
+        } catch (StateException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+        }
+
+        return $this->redirectToRoute('admin_states_index');
+    }
+
+    /**
+     * Handles edit form rendering and submission
+     *
+     * @AdminSecurity(
+     *     "is_granted('update', request.get('_legacy_controller'))",
+     *     redirectRoute="admin_states_index"
+     * )
+     *
+     * @param int $stateId
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function editAction(int $stateId, Request $request): Response
+    {
+        // todo: replace when https://github.com/PrestaShop/PrestaShop/pull/15071 is merged
+        return $this->redirect(
+            $this->getAdminLink(
+                'AdminStates',
+                ['updatestate' => '', 'id_state' => $stateId],
+                true)
+        );
+    }
+
+    /**
+     * Show "Add new" form and handle form submit.
+     *
+     * @AdminSecurity(
+     *     "is_granted(['create'], request.get('_legacy_controller'))",
+     *     redirectRoute="admin_states_index",
+     *     message="You do not have permission to create this."
+     * )
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request)
+    {
+        // todo: replace when https://github.com/PrestaShop/PrestaShop/pull/15071 is merged
+        return $this->redirect(
+            $this->getAdminLink(
+                'AdminStates',
+                ['addstate' => ''],
+                true)
+        );
+    }
+
+    /**
+     * Toggles state status
+     *
+     * @AdminSecurity("is_granted('update', request.get('_legacy_controller'))", redirectRoute="admin_states_index")
+     * @DemoRestricted(redirectRoute="admin_states_index")
+     *
+     * @param int $stateId
+     *
+     * @return RedirectResponse
+     */
+    public function toggleStatusAction(int $stateId): RedirectResponse
+    {
+        try {
+            /** @var EditableState $editableState */
+            $editableState = $this->getQueryBus()->handle(new GetStateForEditing($stateId));
+            $this->getCommandBus()->handle(
+                new ToggleStateStatusCommand((int) $stateId, !$editableState->isActive())
```

Comment:
See https://github.com/PrestaShop/PrestaShop/issues/17407

Reply:
@matks So, it means, there is no need to pass the actual status to the command?